### PR TITLE
Better FreeBSD distribution facts

### LINKED ISSUE #15865: Better FreeBSD distribution facts
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /home/ale/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

FreeBSD 10.2
##### SUMMARY

To write generic roles is fundamental to detect operating system version and release information.
The distribution_version does not show coherent version information on FreeBSD, additionally there's no distribution_major_version variable set.

##### STEPS TO REPRODUCE

$ ansible -i ./hosts -m setup local | grep distri
##### EXPECTED RESULTS

Having distribution_version coherent and ansible_distribution_major_version set makes possible to use this facts on generic tasks that works for linux and FreeBSD.

```
$ ansible -i ./hosts -m setup local | grep distri
        "ansible_distribution": "FreeBSD", 
        "ansible_distribution_major_version": "10", 
        "ansible_distribution_release": "10.2-RELEASE-p9", 
        "ansible_distribution_version": "10.2", 
```
##### ACTUAL RESULTS

The distribution_version is not quite usable, it should be a number as they are in the other platforms.

```
$ ansible -i ./hosts -m setup local | grep distri
        "ansible_distribution": "FreeBSD", 
        "ansible_distribution_release": "10.2-RELEASE-p9", 
        "ansible_distribution_version": "FreeBSD 10.2-RELEASE-p9 #0: Thu Jan 14 01:32:46 UTC 2016     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC", 
```

### PR DESCRIPTION
**Pull request by biancalana**
##### ISSUE TYPE

- Bugfix Pull Request
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /home/ale/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Fixes #15865 
Make distribution_version coherent and set distribution_major_version

Before:

```
"ansible_distribution": "FreeBSD", 
        "ansible_distribution_release": "10.2-RELEASE-p9", 
        "ansible_distribution_version": "FreeBSD 10.2-RELEASE-p9 #0: Thu Jan 14 01:32:46 UTC 2016     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC"
```

After:

```
"ansible_distribution": "FreeBSD", 
        "ansible_distribution_major_version": "10", 
        "ansible_distribution_release": "10.2-RELEASE-p9", 
        "ansible_distribution_version": "10.2",
```

### CODE REVIEW

**Comment by mscherer** [comment]
CI si broken since 0cb05d8ac994ec it seems. As a side note, this conflict with my changes: https://github.com/ansible/ansible/pull/15833 , but I guess I could add the same support for getting better netbsd support when we see which one is merged first.

File: lib/ansible/module_utils/facts.py [hunk 1]
```diff
@@ -757,6 +757,13 @@ def get_distribution_Darwin(self):
         data = out.split()[-1]
         self.facts['distribution_version'] = data
 
+    def get_distribution_FreeBSD(self):
+        self.facts['distribution_release'] = platform.release()
+        data = re.search('(\d+)\.(\d+)-RELEASE.*', self.facts['distribution_release'])
+        if data:
+            self.facts['distribution_major_version'] = data.group(1)
+            self.facts['distribution_version'] = '{0}.{1}'.format(data.group(1), data.group(2))
```
Comment (bcoca) [inline lib/ansible/module_utils/facts.py]: format is not python 2.4 friendly

Comment (biancalana) [inline lib/ansible/module_utils/facts.py]: @bcoca I'm not a python dev, what is the better way ?

**Comment by biancalana** [comment]
@mscherer my objetive here was to make facts (ansible_distribution_release, ansible_distribution_major_version and ansible_distribution_version) more close to what you have for linux, so we can leverage that to improve the usability of actual roles/tasks for *BSDs

File: lib/ansible/module_utils/facts.py (hunk 1 above)
Comment (mscherer) [inline lib/ansible/module_utils/facts.py]: Using the % operator. Like:

```
 '%s.%s' % (data.group(1), data.group(2))
```

**Comment by mscherer** [comment]
I think that's a good idea, so please go ahead with the PR

**Comment by robinro** [comment]
what about NetBSD, maybe we can fix all/more of the BSDs together?

**Comment by biancalana** [comment]
@bcoca rebased against devel...

**Pull request merged**